### PR TITLE
[HDInsight] Adding Swagger-generated SDK and tests

### PR DESCRIPTION
**Pull request by wawon-msft**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@wawon-msft 4 tests failed.
Check the Management.HDInsight.Tests.trx
under buildArtifacts\Testoutput\netcore
and you will see which tests failed.

**Comment by wawon-msft** [comment]
Thanks for the tip. The tests have been fixed.

File: src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.HDInsight</PackageId>
+    <Description>Azure HDInsight Management SDK Library</Description>
+    <AssemblyName>Microsoft.Azure.Management.HDInsight</AssemblyName>
+    <VersionPrefix>3.1.0-preview</VersionPrefix>
+    <PackageTags>Microsoft Azure HDInsight Management;HDInsight;HDInsight Management</PackageTags>
```
Comment (shahabhijeet) [inline src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj]: @wawon-msft missing PackageReleaseNotes, see e.g.
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj#L10

File: src/SDKs/HDInsight/Management.HDInsight/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+[assembly: AssemblyTitle("Microsoft Azure HDInsight Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure HDInsight.")]
+[assembly: AssemblyVersion("3.1.0.0")]
+[assembly: AssemblyFileVersion("3.1.0.0")]
+
+[assembly: InternalsVisibleTo("Microsoft.Azure.Management.HDInsight.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```
Comment (shahabhijeet) [inline src/SDKs/HDInsight/Management.HDInsight/Properties/AssemblyInfo.cs]: @wawon-msft any reason you need this?
What scenarios do you have for which you need internals visible? Are you writing API tests only?

File: src/SDKs/HDInsight/Management.HDInsight.Tests/Management.HDInsight.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  <ItemGroup>
+    <ProjectReference Include="..\Management.HDInsight\Management.HDInsight.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="SessionRecords\Management.HDInsight.Tests.CapabilitiesTests\TestGetCapabilities.json">
```
Comment (shahabhijeet) [inline src/SDKs/HDInsight/Management.HDInsight.Tests/Management.HDInsight.Tests.csproj]: @wawon-msft replace all these entries as below
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/BatchAI.Tests/BatchAI.Tests.csproj#L18

Also you need to use 1.6.0 or greater version of Resource manager for recording all your tests. You will need to add 1.6.0 version as a nuget reference and use it. The one that is by default added is 1.1.0 which is really old and should not be used.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/HDInsight/Management.HDInsight/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (wawon-msft) [reply src/SDKs/HDInsight/Management.HDInsight/Properties/AssemblyInfo.cs]: There is a helper method in CreateParametersConverter called GetNodeSize, which to my understanding is the only time we requires access to internals.

**Comment by shahabhijeet** [comment]
@wawon-msft the comments have not been addressed in a week. Will close this PR for now, once you have it addressed you can always reopen the same PR or if you feel you can open a new PR with the changes.

**Comment by wawon-msft** [comment]
@shahabhijeet Sorry for the inactivity, I'll open a new pull request.

File: src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj (hunk 1 above)
Comment (wawon-msft) [reply src/SDKs/HDInsight/Management.HDInsight/Management.HDInsight.csproj]: Resolved in PR #3845.

File: src/SDKs/HDInsight/Management.HDInsight.Tests/Management.HDInsight.Tests.csproj (hunk 3 above)
Comment (wawon-msft) [reply src/SDKs/HDInsight/Management.HDInsight.Tests/Management.HDInsight.Tests.csproj]: Resolved in PR #3845.

**Pull request closed without merging**